### ISSUE TITLE
DemuxFastq CLI parsing bug for --quality-format

### ISSUE THREAD

**Issue by gshiba** [fulcrumgenomics/fgbio#407]
`DemuxFastq` fails when `-q` is used on command line.

Set up:

```
$ zcat r1.fastq.gz
@M00001:001:000000000-AAAAA:1:1101:16729:1382 1:N:0:1
TCGTGACAGGAATCAAATGAAAACACTTGGT
+
1>1>11AFF?FFGGGGGGCBDGBGCFGH11B

$ zcat r2.fastq.gz
@M00001:001:000000000-AAAAA:1:1101:16729:1382 2:N:0:1
TCGTGACAGGAATCAAATGAAAACACTTGGT
+
1>1>11AFF?FFGGGGGGCBDGBGCFGH11B

$ cat samples.csv
Sample_Barcode,Sample_ID,Sample_Name,Library_ID,Description
TCGTTCGT,foo,S1,,

$ fgbio -v  # conda installation
Version: 0.6.1
```

Works fine if tool auto detects:
```
$ fgbio DemuxFastqs -i r1.fastq.gz r2.fastq.gz -o demuxed --read-structures 4B+T 4B+T --metadata samples.csv
[2018/06/01 12:11:27 | DemuxFastqs | Info] Assuming input metadata file is simple CSV file.
[2018/06/01 12:11:27 | FgBioMain | Info] Executing DemuxFastqs from fgbio version 0.6.1 as gosuke@neptune on JRE 1.8.0_121-b15 with snappy
[2018/06/01 12:11:27 | DemuxFastqs | Info] Auto-detected quality format as: Standard
[2018/06/01 12:11:28 | FgBioMain | Info] DemuxFastqs completed. Elapsed time: 0.15 minutes.
```

Fails when I specify a quality format
```
$ fgbio DemuxFastqs -i r1.fastq.gz r2.fastq.gz -o demuxed2 -q Standard --read-structures 4B+T 4B+T --metadata samples.csv
USAGE: fgbio [fgbio arguments] [command name] [command arguments]
Version: 0.6.1

[...]

Argument 'quality-format' could not be constructed from string
        ...Cannot find string ctor for 'QualityEncoding': com.fulcrumgenomics.fastq.QualityEncoding$.apply(java.lang.String)
java.lang.Class.getMethod(Class.java:1786)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.$anonfun$buildUnitFromString$6(ReflectionUtil.scala:405)
scala.util.Failure.getOrElse(Try.scala:218)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.$anonfun$buildUnitFromString$1(ReflectionUtil.scala:399)
scala.util.Try$.apply(Try.scala:209)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.buildUnitFromString(ReflectionUtil.scala:370)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.$anonfun$buildUnitOrOptionFromString$1(ReflectionUtil.scala:364)
scala.util.Try$.apply(Try.scala:209)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.buildUnitOrOptionFromString(ReflectionUtil.scala:363)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.$anonfun$typedValueFromString$1(ReflectionUtil.scala:349)
scala.util.Try$.apply(Try.scala:209)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.typedValueFromString(ReflectionUtil.scala:315)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.$anonfun$constructFromString$1(ReflectionUtil.scala:293)
scala.util.Try$.apply(Try.scala:209)
com.fulcrumgenomics.commons.reflect.ReflectionUtil$.constructFromString(ReflectionUtil.scala:287)
com.fulcrumgenomics.sopt.cmdline.ClpArgument.setArgument(ClpReflectiveBuilder.scala:173)
com.fulcrumgenomics.sopt.cmdline.CommandLineProgramParser.$anonfun$parseArgs$4(CommandLineProgramParser.scala:184)
com.fulcrumgenomics.sopt.cmdline.CommandLineProgramParser.$anonfun$parseArgs$4$adapted(CommandLineProgramParser.scala:184)
scala.Option.foreach(Option.scala:257)
com.fulcrumgenomics.sopt.cmdline.CommandLineProgramParser.$anonfun$parseArgs$3(CommandLineProgramParser.scala:184)
scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
com.fulcrumgenomics.sopt.parsing.OptionParser.$anonfun$foreach$2(OptionParser.scala:85)
scala.collection.immutable.List.foreach(List.scala:389)
com.fulcrumgenomics.sopt.parsing.OptionParser.foreach(OptionParser.scala:84)
scala.collection.TraversableLike.map(TraversableLike.scala:234)
scala.collection.TraversableLike.map$(TraversableLike.scala:227)
com.fulcrumgenomics.sopt.parsing.OptionParser.map(OptionParser.scala:42)
com.fulcrumgenomics.sopt.cmdline.CommandLineProgramParser.parseArgs(CommandLineProgramParser.scala:182)
com.fulcrumgenomics.sopt.cmdline.CommandLineProgramParser.parseAndBuild(CommandLineProgramParser.scala:136)
com.fulcrumgenomics.sopt.cmdline.CommandLineParser.parseSubCommand(CommandLineParser.scala:312)
com.fulcrumgenomics.sopt.cmdline.CommandLineParser.parseCommandAndSubCommand(CommandLineParser.scala:412)
com.fulcrumgenomics.sopt.Sopt$.parseCommandAndSubCommand(Sopt.scala:159)
com.fulcrumgenomics.cmdline.FgBioMain.makeItSo(FgBioMain.scala:89)
com.fulcrumgenomics.cmdline.FgBioMain.makeItSoAndExit(FgBioMain.scala:80)
com.fulcrumgenomics.cmdline.FgBioMain$.main(FgBioMain.scala:48)
com.fulcrumgenomics.cmdline.FgBioMain.main(FgBioMain.scala)
```

**Comment by nh13** [comment]
Thanks for the report, it looks like a bug to me, I'll check it out.

**Comment by nh13** [comment]
I have a fix in #417 if you want to check it out.
